docs/customizing.md

```diff
@@ -0,0 +1,238 @@
+   of pulpcore (3.12.2).
+1. Confirm that all the latest stable releases of your desired plugins are compatible with pulpcore
+   3.12.2, such as by reading the release announcement email thread for pulpcore 3.12.1, reading the
+plugins README, or as a last resort, reading their `setup.py`.
+1. Set `pulp_install_plugins` with each plugin listed as a key, with an empty enum '{}' as each plugin's value.
+1. Run `pulp_installer`.
+1. Make sure to save your variables/playbook for later usage.
+
+Example `pulp_install_plugins`:
+```
+  vars:
+    pulp_install_plugins:
+      pulp-container: {}
+      pulp-file: {}
+      pulp-rpm: {}
+```
+
+Upgrading your installation:
+
+1. Observe what is the latest version of `pulp_installer`, and what version of pulpcore it installed
+   (3.12.2).
+1. Confirm that all the latest stable releases of **currently installed** plugins are compatible
+   with pulpcore 3.12.2, such as by reading the release announcement email thread for pulpcore 3.12.1,
+reading the plugins README, or as a last resort, reading their setup.py.
+1. If they are not all compatible yet, **wait** for the plugins to be updated for
+   compatibility.
+1. Upgrade `pulp_installer` to the latest version.
+1. Set `pulp_install_plugins` with each plugin listed as a key, and with each plugin having a key under it called `upgrade` set to the value `true`.
+1. Re-run `pulp_installer`.
+1. Make sure to save your variables/playbook for later usage.
+
+Example `pulp_install_plugins`:
+```
+  vars:
+    pulp_install_plugins:
+      pulp-container:
+         upgrade: true
+      pulp-file:
+         upgrade: true
+      pulp-rpm:
+         upgrade: true
+```
+
+### Specifying Exact Versions with Reproducibility:
+
+Initial installation:
+
+1. Observe the latest branch of `pulp_installer`, and what version of pulpcore it installs (3.12.2).
+1. Confirm that all the latest stable releases of your desired plugins are compatible with pulpcore
+   3.12.2, such as by reading the release announcement email thread for pulpcore 3.12.1, reading the
+plugins README, or as a last resort, reading their setup.py.
+1. If they are not all compatible yet, try the last version of the installer that installs pulpcore
+   3.11.z . Then confirm that there exist stable releases of your desired plugins that are compatible
+with pulpcore 3.11.z. If there are none, try pulpcore 3.10.z, and repeat.
+1. Once a compatible pulpcore version is found, Set `pulp_install_plugins` with each plugin listed as a key, and with each plugin having a key under it called `version` set to the version as the value (with quotes).
+1. Run `pulp_installer`
+1. Make sure to save your variables/playbook for later usage.
+
+Example `pulp_install_plugins` (with bogus version values):
+```
+  vars:
+    pulp_install_plugins:
+      pulp-container:
+         version: "4.5.6"
+      pulp-file:
+         version: "5.6.7"
+      pulp-rpm:
+         version: "6.7.8"
+```
```

Comment:
I think, this should include the `pulp_version` variable.

Reply:
We do not want users ever setting that though. The pulp_version is assumed to be specific to the installer version by default.